### PR TITLE
fix/logs: saved queries fixes

### PR DESCRIPTION
**Pull request by Ziinc**
image

This PR adds filtering logic for saved queries, other query types are no longer saved. Once merged, can flip the feature flag.

@joshenlim @MildTomato @phamhieu  RE project content stores:
Is it expected that the api returns the content array on the `data` key? if so, then the implementation of the ProjectContentStore does not make sense, as it takes the `Object.values(response)` and attempts to filter on it. The content store assumes that it receives an array of content objects, but in this case it is only receiving one.

This means that the `list()` filtering logic is redundant (since it filters on response values), and I had to work around it in this case. I would assume that this affects other store functions like `sqlSnippets` and `reports` as well. AFAIK there are two different versions of sql editor, and this store is used in the new one.

### CODE REVIEW

**Comment by joshenlim** [comment]
> Is it expected that the api returns the content array on the data key?
ahh i see what you mean now - maybe @phamhieu can chime in on this too, but i don't think the API should be returning the result as `{ data: [] }` actually, but rather just return the array directly for consistency across our APIs

File: studio/pages/project/[ref]/logs-explorer/saved.tsx [hunk 1]
```diff
@@ -9,29 +9,33 @@ import LogsExplorerLayout from 'components/layouts/LogsExplorerLayout/LogsExplor
...
       <LogsExplorerLayout>
-        <Loading active={true}>loading</Loading>
+        <Loading active={true}>{null}</Loading>
       </LogsExplorerLayout>
     )
+  }
 
-  const saved = content.list()
-
+  const saved: UserContent[] = (contentList?.[0] || []).filter((content: UserContent)=> content.type === 'log_sql')
```
Comment (MildTomato) [inline studio/pages/project/[ref]/logs-explorer/saved.tsx:34]: possibly could deal with this in the store itsself.
could just refactor later.



```tsx
  // studio/stores/content/ProjectContentStore.ts
  
  // add this
  logsQueries(filter?: any) {
    let arr = Object.values(this.data)
    if (!!filter) {
      const arr_filtered = arr
        .filter((x) => x.type == 'log_sql')
        .filter(filter)
        .sort((a, b) => a.name.localeCompare(b.name))

      return arr_filtered
    } else {
      const arr_filtered = arr
        .filter((x) => x.type == 'log_sql')
        .sort((a, b) => a.name.localeCompare(b.name))

      return arr_filtered
    }
  }
  
 // in page
 const {content} = useStore()
 await content.load()
 const queries = content.logsQueries()
  ```

Comment (Ziinc) [reply studio/pages/project/[ref]/logs-explorer/saved.tsx:34]: should I wait for @phamhieu 's https://github.com/supabase/supabase/pull/6586 ?

**Comment by phamhieu** [comment]
>Is it expected that the api returns the content array on the data key? if so, then the implementation of the ProjectContentStore does not make sense, as it takes the Object.values(response) and attempts to filter on it.

on fetchData [here](https://github.com/supabase/supabase/blob/e2e534eac154505b33da66f2a35e0f48f761435a/studio/stores/content/ProjectContentStore.ts#L64-L69), we fetch the array items and save it into `data`as dictionary with id as the key. 

>This means that the list() filtering logic is redundant (since it filters on response values)

method list() filters the new `data` dictionary not the response array.

**Comment by Ziinc** [comment]
In any case, the code has been refactored to @MildTomato 's suggestions, the store logic can be refactored later on.

**Review by burggraf — APPROVED** [review]

**Review by MildTomato — APPROVED** [review]

**Review by MildTomato — APPROVED** [review]

**Pull request merged**